### PR TITLE
fix: add Transform implementation for PreCommitInfoExtractorV8

### PR DESCRIPTION
**Pull request by kasteph**
For earlier parts of the chain, i.e. when miner actors were v8 and below, we get the following error:

```
{
    "level": "warn",
    "ts": "2023-07-06T00:03:04.875Z",
    "logger": "lily/tasks",
    "caller": "actorstate/task.go:140",
    "msg": "failed to transform with error: ",
    "error": "expected MinerPreCommitInfoV9List type but got: miner.MinerPreCommitInfoList"
}
```

### CODE REVIEW

**Review by Terryhung — APPROVED** [review]
nice!

**Comment by birdychang** [comment]
Thanks for the fix!

File: storage/sql.go [hunk 1]
```diff
@@ -352,10 +353,13 @@ func verifyModel(ctx context.Context, db *pg.DB, schemaName string, m *orm.Table
...
-		if datatype == "timestamp with time zone" {
+		switch datatype {
+		case "timestamp with time zone":
+			fallthrough
+		case "timestamp without time zone":
 			datatype = "timestamptz"
-		} else if datatype == "timestamp without time zone" {
-			datatype = "timestamp"
+		case "ARRAY":
+			datatype = "bigint[]"
```
Comment (birdychang) [inline storage/sql.go:362]: why is this needed?

File: chain/indexer/integrated/processor/state.go [hunk 2]
```diff
@@ -449,19 +449,28 @@ func MakeProcessors(api tasks.DataSource, indexerTasks []string) (*IndexerProces
 				),
 				minertask.LockedFundsExtractor{},
 			)
+		case tasktype.MinerPreCommitInfoV1_8:
```
Comment (birdychang) [inline chain/indexer/integrated/processor/state.go:452]: Thank you! This is much better.

**Review by birdychang — APPROVED** [review]

File: storage/sql.go (hunk 1 above)
Comment (kasteph) [reply storage/sql.go:362]: Because of this: https://github.com/filecoin-project/lily/blob/master/storage/sql_test.go#L49 -- it's a convenience function being used in such a test.

Moreover, the mapping of data types between Postgres and Golang is not one-to-one [that's why the original code has this condition as well](https://github.com/filecoin-project/lily/blob/master/storage/sql.go#L355-L359) because it tries to equate a Postgres datatype with a `fld.SQLType` [here](https://github.com/filecoin-project/lily/blob/master/storage/sql.go#L361-L363)
